twitter/twitter.php

```diff
@@ -173,6 +174,39 @@ function twitter_follow(App $a, array &$contact)
 	}
 }
 
+function twitter_unfollow(App $a, array &$hook_data)
+{
+	$contact = $hook_data['contact'];
+	Logger::info('Check if contact is twitter contact', ['url' => $contact["url"]]);
+
+	if (!strstr($contact["url"], "://twitter.com") && !strstr($contact["url"], "@twitter.com")) {
+		return;
+	}
+
+	// contact seems to be a twitter contact, so continue
+	$nickname = preg_replace("=https?://twitter.com/(.*)=ism", "$1", $contact["url"]);
+	$nickname = str_replace("@twitter.com", "", $nickname);
```

Comment:
We don't need to extract the nickname this way, we have got a field for that. I guess that we have to clean up the code for places where the nickname is still extracted that way.